File: AccuPay/Domain/TimeLogs/TimeAttendanceLog.vb
Option Strict On
Imports System.ComponentModel.DataAnnotations
Imports System.ComponentModel.DataAnnotations.Schema

Namespace Global.AccuPay.Entity

    <Table("employeetimeattendancelog")>
    Public Class TimeAttendanceLog

        <Key>
        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property RowID As Integer?

        Public Property OrganizationID As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Identity)>
        Public Property Created As Date

        Public Property CreatedBy As Integer?

        <DatabaseGenerated(DatabaseGeneratedOption.Computed)>
        Public Property LastUpd As Date?

        Public Property LastUpdBy As Integer?

        Public Property ImportNumber As String

        Public Property TimeStamp As Date

        Public Property IsTimeIn As Boolean?

        Public Property WorkDay As Date

        Public Property EmployeeID As Integer

        <ForeignKey("EmployeeID")>
        Public Overridable Property Employee As Employee

        Public ReadOnly Property IsTimeInDescription() As String
            Get
                If IsTimeIn Is Nothing Then Return ""

                Return If(IsTimeIn = True, "IN", "OUT")
            End Get
        End Property

    End Class


End Namespace

File: Accupay.Data/Entities/TimeAttendanceLog.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AccuPay.Data.Entities
{
    [Table("employeetimeattendancelog")]
    public class TimeAttendanceLog
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int? RowID { get; set; }

        public int? OrganizationID { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime Created { get; set; }

        public int? CreatedBy { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime? LastUpd { get; set; }

        public int? LastUpdBy { get; set; }

        public string ImportNumber { get; set; }

        public DateTime TimeStamp { get; set; }

        public bool? IsTimeIn { get; set; }

        public DateTime WorkDay { get; set; }

        public int EmployeeID { get; set; }

        [ForeignKey("EmployeeID")]
        public virtual Employee Employee { get; set; }

        public string IsTimeInDescription
        {
            get
            {
                if (IsTimeIn == null)
                    return "";

                return IsTimeIn == true ? "IN" : "OUT";
            }
        }
    }
}